File: app/controllers/tasks_controller.rb
class TasksController < ApplicationController

  def new
    @todolist = Todolist.find(params[:todolist_id])
  end

  def create
    @task = Task.new(task_params)
    @todolist = Todolist.find(params[:todolist_id])
    @task.todolist = @todolist
    @project = Project.find(params[:project_id])
    if @task.save

      redirect_to project_todolist_path(@project, @todolist)
    else
      render :new
    end
  end

  def edit
    @task = Task.find(params[:id])
  end

  def update
    @task = Task.find(params[:id])
    @project = Project.find(params[:project_id])
    @todolist = Todolist.find(params[:todolist_id])

    @task.update(status: params[:status])
    redirect_to project_todolist_path(@project, @todolist)
  end

  private

  def task_params
    params.require(:task).permit(:description, :status)
  end

end

File: app/javascript/controllers/tasks_controller.js
import { Controller } from "@hotwired/stimulus"

export default class extends Controller {

    update(e) {
        const id = e.target.dataset.id
        const project_id = e.target.dataset.project
        const todolist_id = e.target.dataset.todolist
        const csrfToken = document.querySelector("[name='csrf-token']").content
        const url = `/projects/${project_id}/todolists/${todolist_id}/tasks/${id}`
        this.element.outerHTML = ""

        fetch(url, {
            method: 'PATCH', // *GET, POST, PUT, DELETE, etc.
            headers: {
                'Content-Type': 'application/json',
                'X-CSRF-Token': csrfToken
            },
            body: JSON.stringify({ status: e.target.checked }) // body data type must match "Content-Type" header
        })
    }
}
